### ISSUE TITLE
Reproducible Builds

### ISSUE THREAD

**Issue by IzzySoft** [patzly/tack-android#41]
At IzzyOnDroid we support [Reproducible Builds](https://reproducible-builds.org/) (see: [Reproducible Builds, special client support and more at IzzyOnDroid](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients)). Trying for yours, I was able to successfully generate the APK using `./gradlew assembleRelease`, but the resulting APKs were not identical. The problem seem to be some of the PNG files:

```diff
  -rw----     0.0 fat      524 b-      256 defN 1981-01-01 01:01:02 5f5f8ecc res/By.xml
- -rw----     0.0 fat     5100 b-     5100 stor 1981-01-01 01:01:02 7ff56278 res/CH.png
+ -rw----     0.0 fat     5093 b-     5093 stor 1981-01-01 01:01:02 1153a17a res/CH.png
  -rw----     0.0 fat      244 b-      244 stor 1981-01-01 01:01:02 bde23956 res/CK.9.png
  -rw----     0.0 fat      536 b-      267 defN 1981-01-01 01:01:02 84fc072b res/CQ.xml
  -rw----     0.0 fat     1088 b-      403 defN 1981-01-01 01:01:02 fbf64bcb res/CX.xml
@@ -558,7 +558,7 @@
  -rw----     0.0 fat    30452 b-    30452 stor 1981-01-01 01:01:02 c44d44f3 res/al.wav
  -rw----     0.0 fat      712 b-      361 defN 1981-01-01 01:01:02 4f284a1e res/ao.xml
  -rw----     0.0 fat      391 b-      391 stor 1981-01-01 01:01:02 4fcf3d5b res/ar.png
- -rw----     0.0 fat     1199 b-     1199 stor 1981-01-01 01:01:02 81db17cb res/bB.png
+ -rw----     0.0 fat     1198 b-     1198 stor 1981-01-01 01:01:02 bab0dc6c res/bB.png
  -rw----     0.0 fat     1408 b-      426 defN 1981-01-01 01:01:02 3d1151c4 res/bL.xml
  -rw----     0.0 fat     2084 b-      860 defN 1981-01-01 01:01:02 a5d45afb res/bO.xml
  -rw----     0.0 fat      880 b-      418 defN 1981-01-01 01:01:02 c3130486 res/bT.xml
@@ -1072,7 +1072,7 @@
  -rw----     0.0 fat      592 b-      317 defN 1981-01-01 01:01:02 c844c25d res/ya.xml
  -rw----     0.0 fat     1224 b-      496 defN 1981-01-01 01:01:02 8ca9ebd7 res/yd.xml
  -rw----     0.0 fat      395 b-      395 stor 1981-01-01 01:01:02 972e3ff3 res/yg.9.png
- -rw----     0.0 fat     1740 b-     1740 stor 1981-01-01 01:01:02 45664730 res/yi.png
+ -rw----     0.0 fat     1743 b-     1743 stor 1981-01-01 01:01:02 620b8fb7 res/yi.png
  -rw----     0.0 fat      221 b-      221 stor 1981-01-01 01:01:02 9944f31a res/z-.9.png
```

Could it be you use something like [PNG Crush/Crunch](https://f-droid.org/en/docs/Reproducible_Builds/#png-crushcrunch), or those PNGs are [generated from vector drawables](https://f-droid.org/en/docs/Reproducible_Builds/#pngs-generated-from-vector-drawables) (links go to possible solutions)? Those operations are unfortunately non-deterministic and thus make RB impossible.

We'd appreciate if you could help making your build reproducible. We've prepared some [hints on reproducible builds](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Reproducible-Builds) for that.

Looking forward to your reply!

**Comment by patzly** [comment]
Hi Izzy, I've added the `vectorDrawables.generatedDensities = []` line to the build gradle, could you test it again? For me there weren't any byte size differences anymore at least.

**Comment by IzzySoft** [comment]
To test I need to know which commit, and would need your APK (built from the very same commit) to compare against. You can attach the APK here (just **rename** it to `.zip` so Github allows the attachment, that way I can use the link directly – please do not put it into another zip or that would not work).

**Comment by patzly** [comment]
Here it is, built from commit a9cc2d473bb472e85f4da3cb39ef3875bab5accf: [Tack-Android-v4.4.5.zip](https://github.com/user-attachments/files/17412015/Tack-Android-v4.4.5.zip)

**Comment by IzzySoft** [comment]
Congrats, that's RB! :partying_face: just go ahead: tag that commit, attach that APK, give me a ping, and I establish it with my builder here.

**Comment by IzzySoft** [comment]
@patzly you'll give me a ping when the release is up, will you?

**Comment by patzly** [comment]
Yes, I still need some time for the next version to be finished. Thanks!

**Comment by IzzySoft** [comment]
:crossed_fingers: then!

**Comment by patzly** [comment]
@IzzySoft New release is finally out! Thank you so much and happy new year :)

**Comment by IzzySoft** [comment]
Happy new year to you as well – and congrats! You probably already received your welcome toot: the release is RB, so with the next sync around 7 pm UTC, the green shield will be up :partying_face:
